test-app-reactive/src/main/kotlin/uk/gov/justice/digital/hmpps/hmppstemplatepackagenameasync/service/MessageListener.kt

```diff
@@ -14,7 +15,7 @@ class InboundMessageListener(private val inboundMessageService: InboundMessageSe
   @SqsListener("inboundqueue", factory = "hmppsQueueContainerFactoryProxy")
   fun processMessage(message: Message): CompletableFuture<Void> {
     val event = objectMapper.readValue(message.Message, HmppsEvent::class.java)
-    return CoroutineScope(Dispatchers.Default).future {
+    return CoroutineScope(Context.current().asContextElement()).future {
```

Comment:
So I think I get this - retain the current context or we lose the SpanContext and cannot propogate the Span with the trace id? But it's one of those things a developer would just have to "know". Wondering if we should push it into the library?

Reply:
Yes exactly, we could already be losing the SpanContext wherever we're using coroutines.  This is the provided way to pass the context through, but I can't see a way to do it globally - which means developers just have to "know" as you say.